statediff/service.go

```diff
@@ -147,32 +125,49 @@ func (sds *Service) Loop(chainEventCh chan core.ChainEvent) {
 	}
 }
 
-// processStateDiff method builds the state diff payload from the current and parent block before sending it to listening subscriptions
-func (sds *Service) processStateDiff(currentBlock, parentBlock *types.Block) error {
-	stateDiff, err := sds.Builder.BuildStateDiff(parentBlock.Root(), currentBlock.Root(), currentBlock.Number(), currentBlock.Hash())
-	if err != nil {
-		return err
+func (sds *Service) processStateChanges(stateChangeEvent core.StateChangeEvent) error {
+	var accountDiffs []AccountDiff
+	modifiedAccounts := stateChangeEvent.StateChanges.ModifiedAccounts
+	for addr, modifiedAccount := range modifiedAccounts {
+		accountBytes, err := rlp.EncodeToBytes(modifiedAccount.Account)
+		if err != nil {
+			return err
+		}
+
+		var storageDiffs []StorageDiff
+		for k, v := range modifiedAccount.Storage {
+			diff := StorageDiff{
+				Key:   k[:],
+				Value: v[:],
+			}
+			storageDiffs = append(storageDiffs, diff)
+		}
+
+		address := addr
+		a := AccountDiff{
+			Key:     address[:],
+			Value:   accountBytes,
+			Storage: storageDiffs,
+		}
+
+		accountDiffs = append(accountDiffs, a)
 	}
+
+	stateDiff := StateDiff{
+		BlockNumber:     stateChangeEvent.Block.Number(),
+		BlockHash:       stateChangeEvent.Block.Hash(),
+		UpdatedAccounts: accountDiffs,
+		encoded:         nil,
+		err:             nil,
```

Comment:
Are these keys used elsewhere? Wondering if we can remove them from the struct

Reply:
I'm not entirely sure how/where we are using them yet, but I agree if they're no longer used, it would make sense to remove them.